deployment/terraform/assets.go

```diff
@@ -39,6 +40,56 @@ func MustAssetString(name string) string {
 
 // RestoreAssets writes an embedded asset to the given directory
 func RestoreAssets(dir, name string) error {
+	assetPath := "assets/" + name
+
+	// Check if it's a directory
+	fileInfo, err := fs.Stat(assetsFS, assetPath)
+	if err != nil {
+		return err
+	}
+
+	if fileInfo.IsDir() {
+		// It's a directory, copy recursively using WalkDir
+		return fs.WalkDir(assetsFS, assetPath, func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			// Calculate the relative path from the asset root
+			relPath, err := filepath.Rel(assetPath, path)
+			if err != nil {
+				return err
+			}
+
+			// Skip the root directory itself
+			if relPath == "." {
+				return nil
+			}
+
+			targetPath := filepath.Join(dir, name, relPath)
+
+			// Check if it's a directory and only proceed for files, since files are the only ones we want to copy
+			// and we make sure the target directory exists before copying. This prevents empty folders.
+			if !d.IsDir() {
+				// Copy file
+				data, err := assetsFS.ReadFile(path)
+				if err != nil {
+					return err
+				}
+
+				// Create parent directory if it doesn't exist
+				if err := os.MkdirAll(filepath.Dir(targetPath), 0755); err != nil {
+					return err
+				}
+
+				return os.WriteFile(targetPath, data, 0644)
```

Comment:
This code block and the one for copying it if the required asset is a file is virtually the same. Could we refactor into a util function to use it in both places?